### PR TITLE
Improve JavaDocs 

### PR DESCRIPTION
**Pull request by arturobernalg**
Jus add a bunch of javadoc improvement in order to make it more readable when the user search information.

### CODE REVIEW

**Comment by arturobernalg** [comment]
BTW sorry the size of the PR. But IMO it was necessary

File: src/main/java/org/apache/commons/lang3/CharSetUtils.java [hunk 1]
```diff
@@ -45,8 +45,8 @@ public class CharSetUtils {
      * </pre>
      *
      * @see CharSet#getInstance(String...) for set-syntax.
-     * @param str  String to look for characters in, may be null
-     * @param set  String[] set of characters to identify, may be null
+     * @param str  String to look for characters in, may be {@code null}
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/CharSetUtils.java:48]: Is there a reason why "String" is sometimes linked and sometimes not?

**Comment by garydgregory** [comment]
You really need to build the project locally before you create the PR, see the Javadoc errors.

Aside from that, I'm starting to think that this is really making the code harder to read and that it would be better handled by a Doclet.

File: src/main/java/org/apache/commons/lang3/CharSetUtils.java (hunk 1 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/CharSetUtils.java:48]: No Reason at all.

**Comment by arturobernalg** [comment]
HI @garydgregory 

> You really need to build the project locally before you create the PR, see the Javadoc errors. 
> 
> Aside from that, I'm starting to think that this is really making the code harder to read and that it would be better handled by a Doclet.

I lose track of what apache jenkings does. I do not have access

`[INFO] Configuration changed, re-generating javadoc.
[WARNING] Javadoc Warnings
[WARNING] warning: URL https://docs.oracle.com/javase/8/docs/api/element-list was redirected to https://docs.oracle.com/en/java/javase/19/ -- Update the command-line options to suppress this warning.
[WARNING] 1 warning
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  03:28 min
[INFO] Finished at: 2022-09-24T21:01:26+02:00
[INFO] ------------------------------------------------------------------------
 abernal@Arturo-MacBook-Pro  ~/IdeaProyect/commons-lang   feature/better_javadoc `

> You really need to build the project locally before you create the PR, see the Javadoc errors.

None. 

> 
> Aside from that, I'm starting to think that this is really making the code harder to read and that it would be better handled by a Doclet.

**Comment by garydgregory** [comment]
I'm not sure what Jenkins has to do with this. In any case, you can see the failure here:

https://github.com/apache/commons-lang/actions/runs/3119069043/jobs/5058792919

**Comment by aherbert** [comment]
FYI: This javadoc warning:
```
[WARNING] Javadoc Warnings
[WARNING] warning: URL https://docs.oracle.com/javase/8/docs/api/element-list was
redirected to https://docs.oracle.com/en/java/javase/19/
 -- Update the command-line options to suppress this warning.
```
Is due to the `<links>` added to the `maven-javadoc-plugin` in commons-parent for JDK classes. This seems to not be required when building on recent JDK LTS versions (tested on 8, 11, and 17). Only JDK 17 puts out this warning as it attempts to look up an element-list and not a package-list for the javadoc at the provided URL.

Note: To test locally you simply run maven with the default goal: `mvn`. Most of the commons GH actions builds are configured to use the default goal.

**Comment by arturobernalg** [comment]
So do I have to change something? should I close the pr?

**Comment by garydgregory** [comment]
97 files changed... I should be able to take a look over the weekend.

File: src/main/java/org/apache/commons/lang3/BitField.java [hunk 2]
```diff
@@ -56,15 +56,15 @@
  * <p>Flags and data can be retrieved from the integer:</p>
  *
  *<pre>
- *    // Prints true if red, green or blue is non-zero
+ *    // Prints {@code true} if red, green or blue is non-zero
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/BitField.java:59]: This is a code comment, not a Javadoc comment, so using Javadoc decorations is wrong. Please review all the files and revert this type of change please.

**Review by garydgregory — CHANGES_REQUESTED** [review]

File: src/main/java/org/apache/commons/lang3/BitField.java (hunk 2 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/BitField.java:59]: done. Looks like only was this mistake.

Comment (garydgregory) [reply src/main/java/org/apache/commons/lang3/BitField.java:59]: @arturobernalg 
You need to update the PR...

Comment (arturobernalg) [reply src/main/java/org/apache/commons/lang3/BitField.java]: ups. Sorry @garydgregory 
it's done.

File: src/main/java/org/apache/commons/lang3/BitField.java [hunk 3]
```diff
@@ -281,7 +281,7 @@ public byte setByte(final byte holder) {
     }
 
     /**
-     * Sets a boolean BitField.
+     * Sets a {@code boolean} BitField.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/BitField.java:284]: Hi @arturobernalg 
In this case, the bit field is a `short`, not a `boolean`, right? It's the bits in the bit field that represent booleans. I think If we are going to touch Javadocs we should make sure they are correct first ;-) See the other setters as well.

**Pull request closed without merging**